src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -410,50 +379,30 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
     return bailoutOnAlreadyFinishedWork(current, workInProgress);
   }
 
-  function updateHostComponent(current, workInProgress) {
+  function updateHostComponent(current, workInProgress, renderPriority) {
     pushHostContext(workInProgress);
 
     if (current === null) {
       tryToClaimNextHydratableInstance(workInProgress);
     }
 
-    let nextProps = workInProgress.pendingProps;
     const type = workInProgress.type;
-    const prevProps = current !== null ? current.memoizedProps : null;
     const memoizedProps = workInProgress.memoizedProps;
+    let nextProps = workInProgress.pendingProps;
+    if (nextProps === null) {
+      nextProps = memoizedProps;
+      invariant(
+        nextProps !== null,
+        'We should always have pending or current props. This error is ' +
+          'likely caused by a bug in React. Please file an issue.',
+      );
+    }
+    const prevProps = current !== null ? current.memoizedProps : null;
+
     if (hasContextChanged()) {
       // Normally we can bail out on props equality but if context has changed
       // we don't do the bailout and we have to reuse existing props instead.
-      if (nextProps === null) {
-        nextProps = memoizedProps;
-        invariant(
-          nextProps !== null,
-          'We should always have pending or current props. This error is ' +
-            'likely caused by a bug in React. Please file an issue.',
-        );
-      }
     } else if (nextProps === null || memoizedProps === nextProps) {
-      if (
-        !useSyncScheduling &&
-        shouldDeprioritizeSubtree(type, memoizedProps) &&
-        workInProgress.pendingWorkPriority !== OffscreenPriority
-      ) {
-        // This subtree still has work, but it should be deprioritized so we need
-        // to bail out and not do any work yet.
-        // TODO: It would be better if this tree got its correct priority set
-        // during scheduleUpdate instead because otherwise we'll start a higher
-        // priority reconciliation first before we can get down here. However,
-        // that is a bit tricky since workInProgress and current can have
-        // different "hidden" settings.
-        let child = workInProgress.progressedChild;
-        while (child !== null) {
-          // To ensure that this subtree gets its priority reset, the children
-          // need to be reset.
-          child.pendingWorkPriority = OffscreenPriority;
-          child = child.sibling;
-        }
-        return null;
-      }
```

Comment:
I think it's not needed because we handle this during priority reset, here: https://github.com/acdlite/react/blob/9713bd40a6e4eb406b6f66b8a55c4bbd662d5ae3/src/renderers/shared/fiber/ReactFiberScheduler.js#L558-L564